jax/numpy/lax_numpy.py

```diff
@@ -2580,6 +2580,26 @@ def diag(v, k=0):
   else:
     raise ValueError("diag input must be 1d or 2d")
 
+_SCALAR_VALUE_DOC="""\
+Unlike numpy's implementation of diagflat, this does not
+handle scalar values differently
```

Comment:
Maybe something more specific like `"This differs from np.diagflat for some scalar values of v: it always returns a two-dimensional array, where numpy may return a scalar depending on the type of v."`

Reply:
I see :)

Reply:
Added description